src/iam/certhandler/modules/certmodule.cpp

```diff
@@ -0,0 +1,418 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "aos/iam/modules/certmodule.hpp"
+#include "../log.hpp"
+#include "aos/common/tools/memory.hpp"
+
+namespace aos {
+namespace iam {
+namespace certhandler {
+
+/***********************************************************************************************************************
+ * Public
+ **********************************************************************************************************************/
+
+CertModule::CertModule(const String& certType, const ModuleConfig& config)
+    : mCertType(certType)
+    , mModuleConfig(config)
+{
+}
+
+Error CertModule::Init(crypto::x509::ProviderItf& x509Provider, HSMItf& hsm, StorageItf& storage)
+{
+    mX509Provider = &x509Provider;
+    mHSM = &hsm;
+    mStorage = &storage;
+
+    if (mModuleConfig.mSkipValidation) {
+        LOG_WRN() << "Skip validation. type=" << GetCertType();
+
+        return Error::Enum::eNone;
+    }
+
+    auto validCerts = MakeUnique<ModuleCertificates>(&mAllocator);
+    if (!validCerts) {
+        return AOS_ERROR_WRAP(Error::Enum::eNoMemory);
+    }
```

Comment:
Currently it's not clear about memory allocation error handling: return an error / use assert. Current impl follows an approach with PushBack, ie - return an error.

Reply:
The idea is: with static allocator you have preallocated memory and it is almost used by this particular class that is fully covered with unit tests. And if your allocator memory is not enough it will be identified and fixed with unit testing. That's why it is not mandatory to check out of memory error. Push to array is performed on external for current class/module array and not covered with unit testing. So, in this case checking memory error is mandatory.